tools/CheckMetadata.py

```diff
@@ -0,0 +1,179 @@
+        self.assertIn('Affiliations', self.metadata)
+        affiliations = self.metadata['Affiliations']
+        self.assertTrue(isinstance(affiliations, list))
+        self.assertTrue(
+            [isinstance(affiliation, str) for affiliation in affiliations])
+        # Check for duplicates
+        simplified_affiliations = [
+            simplify_name(affiliation) for affiliation in affiliations
+        ]
+        for affiliation in affiliations:
+            self.assertTrue(
+                simplified_affiliations.count(simplify_name(affiliation)) == 1,
+                f"Duplicate affiliation: {affiliation}")
+
+    def test_authors(self):
+        self.assertIn('Authors', self.metadata)
+        authors = self.metadata['Authors']
+        self.assertTrue(isinstance(authors['Description'], str))
+        authors_core = authors['Core']
+        self.assertTrue(isinstance(authors_core['Description'], str))
+        authors_core_list = authors_core['List']
+        self.assertTrue(isinstance(authors_core_list, list))
+        authors_devs = authors['Developers']
+        self.assertTrue(isinstance(authors_devs['Description'], str))
+        authors_devs_list = authors_devs['List']
+        self.assertTrue(isinstance(authors_devs_list, list))
+        authors_contribs = authors['Contributors']
+        self.assertTrue(isinstance(authors_contribs['Description'], str))
+        authors_contribs_list = authors_contribs['List']
+        self.assertTrue(isinstance(authors_contribs_list, list))
+        all_authors = (authors_core_list + authors_devs_list +
+                       authors_contribs_list)
+
+        # Check alphabetical ordering
+        self.assertEqual(
+            sorted(authors_devs_list,
+                   key=lambda a: a['Name']), authors_devs_list,
+            "'Developers' author list should be ordered alphabetically")
+        self.assertEqual(
+            sorted(authors_contribs_list,
+                   key=lambda a: a['Name']), authors_contribs_list,
+            "'Contributors' author list should be ordered alphabetically")
+
+        # Check all authors
+        def check_name(name):
+            split_name = name.split(', ')
+            self.assertTrue(
+                len(split_name) == 2,
+                f"Name '{name}' should be formatted 'Last name, First name'")
+            last_name, first_name = split_name
+            self.assertTrue(last_name, f"Missing last name: {name}")
+            self.assertTrue(first_name, f"Missing first name: {name}")
+
+        all_author_names_simplified = [
+            simplify_name(author['Name']) for author in all_authors
+        ]
+        for author in all_authors:
+            self.assertIn('Name', author)
+            author_name = author['Name']
+            check_name(author_name)
+            self.assertEqual(
+                all_author_names_simplified.count(simplify_name(author_name)),
+                1, f"Duplicate author: {author_name}")
```

Comment:
The "duplicate affiliation" check above is using a different syntax.  I don't know if it matters, but might as well make these all consistent.